packages/core/src/client.ts

```diff
@@ -55,6 +55,7 @@ export class Client<
   TWebSocketProvider extends WebSocketProvider = WebSocketProvider,
 > {
   config: Partial<ClientConfig<TProvider, TWebSocketProvider>>
```

Comment:
Thoughts on removing `Partial`? Would allow use to remove the `|| x.provider` condition in a few places since `provider` is a required constructor argument and will always be available in `this.config`.